Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialComponent.cpp

```diff
@@ -451,29 +478,44 @@ namespace AZ
         {
             AzToolsFramework::ScopedUndoBatch undoBatch("Generating materials.");
             SetDirty();
+            
+            Data::AssetId modelAssetId;
+            MeshComponentRequestBus::EventResult(modelAssetId, GetEntityId(), &MeshComponentRequestBus::Events::GetModelAssetId);
```

Comment:
(I also tested actors with these changes)

Reply:
I do like having MaterialReceiverRequestBus because it is more explicit about the relationship with the material component. Perhaps we should add a GetModelAssetId function to MaterialReceiverRequestBus and call that one instead?

Reply:
Actually, somewhere along the line of code changes, I stopped using modelAssetId so I can just remove this :)

Reply:
Actually, I removed the use of GetModelAssetId, it isn't needed anymore.